### PR TITLE
Add Expand/Collapse to collection items in data model

### LINKED ISSUE #18517: Add way to collapse collection folders in data-model view
When you have a large amount of collections in your schema, the data model view can become very unwieldy. By adding a way to collapse those folders in the view, this can be alleviated a bit. See https://github.com/directus/directus/discussions/9939#discussioncomment-5833036 for a proposed solution.

## Must Haves

- Allow folders to be collapsed with a toggle button

## Should Have

- Sync collapsed state in localStorage 

## Could Have

n/a

## Won't Have

- Collapsing for folders on the folder view

### PR DESCRIPTION
**Pull request by that1matt**
This adds the ability to collapse/expand collection items that have nested collection items.

It uses local storage to set the key of the collection, and value of true/false.

Fixes #18517 

https://github.com/directus/directus/assets/9679004/fbbffa2d-3399-4203-8be6-c048ad7e34e4

Notes:
- Based on my new experience within Vue 3, I am not sure how to get the `<transition-expand>` animations to apply instead of just closing/opening without any animations.
- I am also not sure if this is the correct way to implement this functionality, there might be a better way, or a more "best practice" option, that I am happy to update to.
- Want to add some test cases, as I am figuring out how to do that.
- I added a new translation string, not sure if this is the correct way to do that. Or if a better string would fit the `Click to Collapse` verbiage.
- I also just picked the two icons that I thought fit, if better ones exist, happy to update them.

### CODE REVIEW

File: app/src/lang/translations/en-US.yaml [hunk 1]
```diff
@@ -66,7 +66,7 @@ indeterminate: Indeterminate
 edit_collection: Edit Collection
 exclusive: Exclusive
 children: Children
-db_only_click_to_configure: 'Database Only: Click to Configure'
```
Comment (that1matt) [inline app/src/lang/translations/en-US.yaml:69]: I am having issues with trying to mimic my local vscode to match the settings for this repo. But without a `.vscode/settings.json` I have been trying to work around it. Not sure if there are suggestions on how to fix these linting/editor formatter issues to match what the Directus repo needs.

Comment (rijkvanzanten) [reply app/src/lang/translations/en-US.yaml]: Do you happen to have the `editorconfig` extension installed? That file should control the basic formatting settings. Everything else is controlled through the auto-linter 🤔

**Review by rijkvanzanten — CHANGES_REQUESTED** [review]
Lets flip the priority! I think it's nicer to have the temp-local-collapse as the primary option, with the "start closed in nav" as a toggle in the collection options. I'll handle that tweak 👍🏻

**Comment by rijkvanzanten** [comment]
Also quick Vue vs React tip: in Vue, it's common to "emit" an event with data going up the tree, rather than passing a callback function down as a prop. Both patterns work, but the emitting is a little more vue-friendly 🙂

**Comment by rijkvanzanten** [comment]
> Based on my new experience within Vue 3, I am not sure how to get the <transition-expand> animations to apply instead of just closing/opening without any animations.

The trick here is that it goes around the element that has a v-if, instead of get the v-if itself. In this case, the whole nested draggable block is de-rendered, so the transition can go around that 👍🏻

**Comment by rijkvanzanten** [comment]
https://github.com/directus/directus/assets/9141017/3a479008-7d76-4c3e-b9c6-3d7f169130cf

Done! Feels great. Thanks for the idea and initial implementation @that1matt! 🚀

**Review by rijkvanzanten — APPROVED** [review]

**Pull request merged**